chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -306,12 +246,32 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
         }
 
         let paid_fee = inputs_total - outputs_total;
-        paid_fee.ok_or(ConnectTransactionError::TxFeeTotalCalcFailed(
+        paid_fee.map(Fee).ok_or(ConnectTransactionError::TxFeeTotalCalcFailed(
             inputs_total,
             outputs_total,
         ))
     }
 
+    fn calculate_coins_total_inputs(
+        &self,
+        inputs: &[TxInput],
+        spender: Spender,
+    ) -> Result<Amount, ConnectTransactionError> {
+        inputs
+            .iter()
+            .map(|input| {
+                self.utxo_cache
+                    .utxo(input.outpoint())
+                    .ok_or_else(|| ConnectTransactionError::DoubleSpendAttempt(spender.clone()))
```

Comment:
The problem here is that calling this double-spend is not really fair or accurate. It may or may not be a double-spend. In bitcoin it's called "InputMissingOrSpent" or something similar. That's more accurate because it could be either.